### PR TITLE
Azure AD - fix events

### PR DESCRIPTION
**Pull request by lvoloshyn-sekoia**
Improvement for https://github.com/SekoiaLab/integration/issues/1068
as well as for https://github.com/SekoiaLab/integration/issues/1074

## Type of Change

- [ ] New intake format
- [ ] Update to existing intake format
- [X] Bug fix
- [ ] Documentation update
- [ ] Other (please describe):

## Affected Intake Formats

- Vendor/product:

### 🔒 CRITICAL - Security Requirements (MUST BE VERIFIED FIRST)

- [X] **No real email addresses** in test files (use: user@example.com, test@example.org)
- [X] **No real passwords or API keys** (use: "P@ssw0rd!", "xxxxxxxxxxxx", "FAKE_TOKEN_123")
- [X] **No real IP addresses** (use TEST-NET ranges: 192.0.2.x, 198.51.100.x, 203.0.113.x)
- [X] **No real phone numbers** (use: +1-555-0100 or similar test numbers)
- [X] **No real names or PII** (use: "John Doe", "Jane Smith", "User123")
- [X] **No real company/organization names** in test data (unless public vendors)
- [X] **No real usernames, hostnames, or domain names** (use: user1, host.example.com, test.corp)
- [X] All test data has been anonymized and verified

### Required for All PRs

- [ ] Code has been linted
    - [ ] with the linter for manifest, smart-descriptions, and test files (`poetry run python linter.py check --changes` in utils/)
    - [ ] with Prettier for the parser code files (`npx prettier --write <module>/<format>/ingest/parser.yml`)
- [ ] All CI/CD checks pass
- [ ] Changes follow the contribution guidelines in `CONTRIBUTING.md`

### Required for New/Updated Intake Formats

- [ ] Clear descriptions provided for modules and formats
- [ ] Logo files included for new modules/formats
- [ ] Parser test coverage is at least 75%
- [ ] At least one smart-description is provided
- [ ] README.md updated (if applicable)
- [ ] Taxonomy mappings are correct and documented
- [ ] Test cases cover edge cases and error handling

### Testing

- [ ] Local tests pass (`poetry run pytest` in utils/)
- [ ] Sample data is representative and **completely anonymized**
- [ ] **No sensitive information** (PII, credentials, real IPs, real emails) in test files
- [ ] Test data uses example.com/example.org domains and TEST-NET IP ranges
- [ ] Parser handles malformed data gracefully

### Documentation

- [ ] Code changes are documented
- [ ] Smart-descriptions are clear and informative
- [ ] Field mappings are explained

## Related Issues

Closes #

## Summary by Sourcery

Fix Azure AD event parsing by using activityDisplayName as a fallback for action names, add mapping for single-user target fields, and include new test fixtures for various user-related events.

Bug Fixes:
- Use activityDisplayName as a fallback when operationName is missing for action.name and event.action

Enhancements:
- Map user.target.name and user.target.id when there is exactly one User targetResource

Tests:
- Add new test fixtures for change_user_password_2, invite_external_user, update_ststoken_2, and update_user events

### CODE REVIEW

**Review by TOUFIKIzakarya — COMMENTED** [review]
Thanks for the fix.
A small comment and can you please apply the anonymization script ?

File: Azure/azure-ad/ingest/parser.yml [hunk 1]
```diff
@@ -195,6 +206,13 @@ stages:
               {%- if prop.displayName == 'Group.DisplayName' -%}{{prop.newValue | from_json or "null"}}{%- endif -%}
             {%- endfor -%}
 
+      - set:
+          azuread.properties.targetServicePrincipalNewDisplayName: >
```
[the remark this answers is not in the document]
Comment (lvoloshyn-sekoia) [reply Azure/azure-ad/ingest/parser.yml]: >And also i can't see targetResources.modifiedProperties[].displayName

isn't it mentioned only as a condition?

I'll change `azuread.properties.targetServicePrincipal.appRole.newValue`, thanks

**Review by TOUFIKIzakarya — APPROVED** [review]
LGFM
